config/api/src/main/java/org/jboss/windup/config/EmptyRuleLifecycleListener.java

```diff
@@ -0,0 +1,50 @@
+package org.jboss.windup.config;
+
+import org.ocpsoft.rewrite.config.Rule;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+/**
+ * This provides a set of default (empty) methods that make it easy to implement {@link RuleLifecycleListener}s that only need to override a subset of
+ * the available methods.
+ * 
+ * @author jsightler
+ *
+ */
+public class EmptyRuleLifecycleListener implements RuleLifecycleListener
```

Comment:
This could just be made abstract and renamed to AbstractRuleLifecycleListener, no?